packages/react-core/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,509 @@
+    this.refCollection[index] = optionRef as HTMLElement;
+  };
+
+  parseTime = (time: string) => {
+    const { delimiter, variant, validateTime } = this.props;
+    let parsedTime = time;
+    let valid = true;
+
+    const date = new Date(time);
+    if (!isNaN(date.getDate())) {
+      const hours =
+        variant === '12hr'
+          ? `${variant === '12hr' && date.getHours() > 11 ? date.getHours() - 12 : date.getHours()}`
+          : `${date.getHours()}`.padStart(2, '0');
+      const minutes = `${date.getMinutes()}`.padStart(2, '0');
+      parsedTime = `${hours}${delimiter}${minutes}${variant === '12hr' ? (date.getHours() > 11 ? 'pm' : 'am') : ''}`;
+    } else {
+      if (validateTime) {
+        valid = validateTime(time);
+      } else {
+        const regexp =
+          variant === '12hr'
+            ? new RegExp(`\\b\\d\\d?${delimiter}[0-5]\\d\\s?([AaPp][Mm])?\\b`)
+            : new RegExp(`\\b\\d\\d?${delimiter}[0-5]\\d\\b`);
+        valid = time === '' || regexp.test(time);
+      }
+      if (
+        variant === '12hr' &&
+        valid &&
+        time !== '' &&
+        !parsedTime.toLowerCase().includes('am') &&
+        !parsedTime.toLowerCase().includes('pm')
+      ) {
+        parsedTime = `${parsedTime}${new Date().getHours() > 11 ? 'pm' : 'am'}`;
+      }
+    }
+
+    this.setState({
+      time: parsedTime,
+      focusedIndex: null,
+      isInvalid: !valid
+    });
+  };
+
+  onToggle = (isOpen: boolean) => {
+    this.setState({
+      isOpen
+    });
+  };
+
+  onSelect = (selection: string, index: number) => {
+    if (this.clearButtonRef && this.clearButtonRef.current) {
+      this.clearButtonRef.current.focus();
+    } else if (this.toggleButtonRef && this.toggleButtonRef.current) {
+      this.toggleButtonRef.current.focus();
+    }
+    this.setState({
+      time: selection,
+      isInvalid: !this.validTimeFormat(selection),
+      isOpen: false,
+      selectedIndex: index,
+      focusedIndex: index,
+      scrollIndex: index
+    });
+  };
+
+  clearSelection = () => {
+    this.setState({
+      time: '',
+      selectedIndex: null,
+      focusedIndex: null,
+      scrollIndex: 0,
+      isOpen: false,
+      isInvalid: false
+    });
+  };
+
+  onInputFocus = (e: React.FocusEvent) => {
+    if (!this.state.isOpen) {
+      this.onToggle(true);
+    }
+    e.stopPropagation();
+  };
+
+  onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const time = event.target.value;
+    if (time === '' || this.validTimeFormat(time)) {
+      if (this.props.onChange) {
+        this.props.onChange(time, event);
+      }
+      this.scrollToSelection(time);
+      this.setState({
+        time,
+        isInvalid: false
+      });
+    } else {
+      this.setState({
+        time,
+        focusedIndex: null
+      });
+    }
+  };
+
+  validTimeFormat = (value: string) => {
+    const regexp = new RegExp(`\\b\\d\\d?${this.props.delimiter}?[0-5]?\\d?\\s?([AaPp][Mm])?\\b`);
```

Comment:
Not sure if this is related to the "need to validate the hours are in range" comment, but this allows hours up to 99:59pm

Reply:
yeppp. gotta clean that up :)